collated/main/A0121261Y.md

```diff
@@ -0,0 +1,856 @@
+            + "(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:0?[1-9]|1\\d|2[0-8])(\\/|-|\\.)(?:(?:0?[1-9]|"
+            + "(?:Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep))|(?:1[0-2]|(?:Oct|Nov|Dec)))"
+            + "\\4(?:(?:1[6-9]|[2-9]\\d)?\\d{2})";
+
+    public static final String TIME_PARSE_FORMAT_CHOICE_12HR = "[h:mma]" + "[h.mma]" + "[hmma]";
+    public static final String TIME_PARSE_FORMAT_CHOICE_24HR = "[k:mm]" + "[k.mm]" + "[kkmm]";
+
+    public static final String[] DATE_PARSE_FORMAT_UNTIMED_CHOICE = {"[d-M-uuuu]","[d-MMM-uuuu]",
+            "[d.M.uuuu]","[d.MMM.uuuu]", "[d/M/uuuu]","[d/MMM/uuuu]","[uuuu-M-dd]","[d-M-yy]",
+            "[d-M-yy]", "[d-MMM-yy]", "[d.M.yy]", "[d.MMM.yy]", "[d/MMM/yy]", "[d/M/yy]"};
+
+    public static final String DATE_TIME_PRINT_FORMAT = "dd-MMM-uuuu h:mma";
+    public static final String DATE_PRINT_FORMAT = "dd-MMM-uuuu";
+    public static final String XML_DATE_TIME_OPTIONAL_FORMAT = "uuuu-MM-dd HH:mm";
+    public static final String TIME_PRINT_FORMAT = "h:mma";
+
+    public final String value; //value to store date in UK format
+    private LocalDateTime startDate; //US format by java YYYY-MM-DD
+    private Optional<LocalDateTime> endDate;
+    private boolean isUntimed;
+
+    /**
+     * Validates given date.
+     *
+     * @param a string consisting of only the date i.e dd-MMM-YYYY no time
+     * @return an untimed date for untimed task
+     * @throws IllegalValueException if given time string is invalid.
+     */
+
+    public Time(String date) throws IllegalValueException {
+        assert date != null;
+        date = fixStoredDataForTest(date);
+        date = date.toUpperCase(); // fix for strings that bypass parser from other components..
+        assert (isValidDate(date)); // if this fails, you have used the wrong constructor
+        endDate = Optional.empty();
+        isUntimed = true;
+        DateTimeFormatter formatter = setDateFormatter();
+        date = fixMonthForJavaFormat(date);
+        this.startDate = LocalDate.parse(date, formatter).atTime(LocalTime.now().truncatedTo(ChronoUnit.MINUTES));
+        value = timeToUkFormat();
+    }
+    /**
+     * Java only accepts camel case dates, this method converts text month into camel case
+     *
+     * @param date
+     * @return date with month in camel case
+     */
+
+    private String fixMonthForJavaFormat(String date) {
+        int capsMonth = -1;
+        date = date.toLowerCase().trim();
+        if (Character.isAlphabetic(date.charAt(2))) { //for 1st character in date with 2 digits 1 delimiter,
+            capsMonth = 2;
+        } else if (Character.isAlphabetic(date.charAt(3))) { // 1st character in date with 1 digit 1 delimiter
+            capsMonth = 3;
+        }
+        if (capsMonth != -1){
+            String fixedMonth = date.substring(0,capsMonth) + Character.toString(date.charAt(capsMonth)).toUpperCase()+date.substring(capsMonth + 1);
```

Comment:
Avoid lines longer than 110 chars. When a line is too long, it requires horizontal scrolling, that slows down the reader/reviewer.